helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -403,6 +463,10 @@ public void invoke(NotificationContext changeContext) throws Exception {
         subscribeForChangesAsyn(changeContext.getType(), _path, _watchChild);
       }
 
+      if (_periodicTriggerInterval > 0) {
```

Comment:
This is not the direct indicator, how about check if _periodicRefreshExecutor != null?